### PR TITLE
Rewrote drawer arround the NavigationView

### LINKED ISSUE #749: No storage is selected when the app starts
No storage is selected when the app starts.

### PR DESCRIPTION
**Pull request by EmmanuelMess**
* Rewrote drawer arround the NavigationView
* Moved the long tab actions to action (Image)Views on the side of the item
* Also moved the logic to another file
* Simplified some code (with member functions)

Fixes #749

### CODE REVIEW

File: app/src/main/java/com/amaze/filemanager/ui/views/drawer/Drawer.java [hunk 1]
```diff
@@ -205,7 +215,8 @@ public void refreshDrawer() {
             synchronized (dataUtils.getServers()) {
                 for (String[] file : dataUtils.getServers()) {
                     addNewItem(menu, SEVERS_GROUP, order++, file[0],
```
Comment (TranceLove) [inline app/src/main/java/com/amaze/filemanager/ui/views/drawer/Drawer.java]: `SERVERS_GROUP` please.

**Comment by EmmanuelMess** [comment]
> The pen beside SFTP connections at the drawer didn't bring up the edit connection dialog, but SMB connections did.

Did you correctly implement long tap functionality previously?

**Comment by EmmanuelMess** [comment]
Also Icon color doesn't change when theme changes.

**Comment by TranceLove** [comment]
> Did you correctly implement long tap functionality previously?

Yes I did.

**Comment by EmmanuelMess** [comment]
>  Yes I did (at least it works with the same gesture). See DrawerAdapter in master.

Missed that one, should be fixed now.

**Comment by EmmanuelMess** [comment]
> The first time I tapped on a SMB connection, it did not connect to server but showed up the SMB URI like smb://<username>:<password>@<smb.host.name> at the location bar. When I tapped other entry at the drawer to go somewhere else and go back to the SMB connection, it succeeded.

Fixed, although it might have some unintended consequences.

**Comment by TranceLove** [comment]
Sorry for the late reply.

After f982ff0 when I tap on a SMB connection it brought up the SMB connection setup dialog. And after I tapped Create it stuck on loading but not connect to the SMB server.

By the way, are we unifying SMB connections and SFTP/SCP connections to use the same icon?

**Comment by EmmanuelMess** [comment]
> After f982ff0 when I tap on a SMB connection it brought up the SMB connection setup dialog. And after I tapped Create it stuck on loading but not connect to the SMB server.

Will retest, but last time it connected...

> By the way, are we unifying SMB connections and SFTP/SCP connections to use the same icon?

Accidental, but we should, unless you have a specific "SFTP/SCP" (not the generic Linux penguin one).

**Comment by EmmanuelMess** [comment]
> After f982ff0 when I tap on a SMB connection it brought up the SMB connection setup dialog. And after I tapped Create it stuck on loading but not connect to the SMB server.

Retested, it works, pressing the pencil will allow you to edit the connection, but not the drawer item.

PS: rebasing to update to master branch

**Comment by TranceLove** [comment]
OK, it's good to go now.

BTW+self note: I suspected the problem I faced with SMB connection mentioned above might really not related to this PR but a glitch I found with saving SMB connections back in the days of #863, when I was looking for references in code. Will get back to it later.

**Pull request merged**